File: test/fixtures/new-logic-lands-without-a-test-dart/lib/session_storage.dart
class StoredSession {
  StoredSession({required this.id});

  final String id;
}

List<StoredSession> loadStoredSessions(String raw) {
  try {
    final ids = _decodeIds(raw);
    return ids.map((id) => StoredSession(id: id)).toList();
  } catch (_) {
    // corrupt-data path — tested below, unlike live_session_controller.dart's siblings.
    return [];
  }
}

List<String> _decodeIds(String raw) {
  if (!raw.startsWith('[') || !raw.endsWith(']')) {
    throw const FormatException('not a json array');
  }
  final inner = raw.substring(1, raw.length - 1);
  return inner.isEmpty ? [] : inner.split(',');
}

Map<String, String> stripForStorage(Map<String, String> session) {
  final rest = Map<String, String>.from(session);
  rest.remove('token');
  return rest;
}

File: test/fixtures/new-logic-lands-without-a-test/sessionStorage.ts
export function loadStoredSessions(raw: string): { id: string }[] {
  try {
    const parsed = JSON.parse(raw);
    return Array.isArray(parsed) ? parsed : [];
  } catch {
    // corrupt-data path — tested below, unlike useLiveSessions.ts's siblings.
    return [];
  }
}

export function stripForStorage(session: { id: string; token: string }) {
  const { token, ...rest } = session;
  return rest;
}
